src/common/load-plugins.js

```diff
@@ -27,7 +27,9 @@ function load(plugins, pluginSearchDirs) {
   }
   // unless pluginSearchDirs are provided, auto-load plugins from node_modules that are parent to Prettier
   if (!pluginSearchDirs.length) {
-    const autoLoadDir = thirdParty.findParentDir(__dirname, "node_modules");
+    const autoLoadDir =
+      findParentDir(__dirname, "node_modules/.pnpm") ||
+      findParentDir(__dirname, "node_modules");
```

Comment:
I don't believe this will work.`node_modules/.pnpm` is the default pnpm virtual store directory, but can be overridden by the `virtual-store-dir` setting in `.npmrc`.  I think a better solution would be to see if the target path exists and is a directory in `src/common/load-plugins.js`:

**Before**
```js
function isDirectory(dir) {
  try {
    return fs.statSync(dir).isDirectory();
  } catch (e) {
    return false;
  }
}
```
**After**
```js
function isDirectory(dir) {
  try {
    return fs.statSync(fs.realpathSync(dir)).isDirectory()
  } catch (e) {
    return false;
  }
}
```

Reply:
Thanks for the information, any suggestion?

Reply:
Not sure. It seems the default path (or whatever is specified in `pluginSearchDirs`) is being resolved relative to `process.cwd()` which sits inside the pnpm virtual store. Packages in the virtual store aren't stored hierarchically so the concept of `findParentDir` does not really work. With pnpm, the package hierarchy is captured in the symlink structure, so the plugin searching will need to use the symlink (not the target) somehow. The only way I have gotten plugins to load is by specifying them in `.prettierrc` using a relative path, e.g. `"plugins": ["./node_modules/prettier-plugin-packagejson"]`

Reply:
I also use [`eslint` plugins](https://eslint.org/docs/user-guide/configuring/plugins) which seem to work fine with `pnpm` ... Looking at the eslint docs/internals it seems they resolve plugins as if you were to `require` them from the config file.

Reply:
Could you accept this PR as is and defer fine tuned configuration. Prettier doesn't work with PNPM

Reply:
Prettier does work with pnpm, it just won’t auto load plugins. You can load them manually in the .prettierrc, for example:  `"plugins": ["./node_modules/prettier-plugin-packagejson"]`